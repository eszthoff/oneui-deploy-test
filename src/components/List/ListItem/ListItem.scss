.ListItem {
    transition: background-color var(--transition-duration);

    &__container {
        display: flex;
        align-items: baseline;
        padding: var(--spacing-3x);
    }

    &:hover {
        background-color: var(--color-hover-background);
    }

    &--isSelected {
        background-color: var(--color-highlight-background);
    }

    &--clickable {
        cursor: pointer;
    }

    &--disabled:hover {
        background-color: var(--color-background);
    }
}
