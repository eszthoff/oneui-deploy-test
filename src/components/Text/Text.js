import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import styles from './Text.scss';

const { block } = bem({
    name: 'Text',
    classnames: styles,
    propsToMods: ['muted']
});

const Text = props => {
    const { children, inline, ...rest } = props;
    const HtmlNodeType = inline ? 'span' : 'p';

    return (
        <HtmlNodeType {...rest} {...block(props)}>
            {children}
        </HtmlNodeType>
    );
};

Text.displayName = 'Text';

Text.propTypes = {
    /** Text should be rendered inline */
    inline: PropTypes.bool,
    /** Render muted (auxiliary) text */
    muted: PropTypes.bool,
    /** Text content */
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired
};

Text.defaultProps = {
    inline: false,
    muted: false
};

export default Text;
