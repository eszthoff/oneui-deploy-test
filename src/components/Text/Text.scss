@import '../../themes/oneui/oneui.scss';

.Text {
    color: var(--color-foreground);
    font: {
        family: var(--font-family-primary);
        size: var(--font-size-normal);
        weight: var(--font-weight-normal);
    }

    &--bold {
        font-weight: var(--font-weight-bold);
    }

    &--muted {
        color: var(--color-muted);
    }

    &--size {
        &_small {
            font-size: var(--font-size-small);
        }

        &_large {
            font-size: var(--font-size-large);
        }
    }
}
