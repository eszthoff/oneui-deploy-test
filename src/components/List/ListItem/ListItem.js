import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import Text from '../../Text';
import styles from './ListItem.scss';

const { block, elem } = bem({
    name: 'ListItem',
    classnames: styles,
    propsToMods: ['isDivided', 'isDividedList', 'isSelected', 'onClick']
});

const ListItem = props => {
    const { children, isDivided, isDividedList, isSelected, onClick, ...rest } = props;
    const customBlockMod = { clickable: typeof onClick === 'function' };

    return (
        <li {...rest} {...block(props, customBlockMod)}>
            <div onClick={onClick} role="presentation" {...elem('container', props)}>
                {React.Children.map(
                    children,
                    child => (typeof child === 'string' ? <Text>{child}</Text> : child)
                )}
            </div>
        </li>
    );
};

ListItem.displayName = 'ListItem';

ListItem.propTypes = {
    /** List Items */
    children: PropTypes.node,
    /** Formats this item as selected  */
    isSelected: PropTypes.bool,
    /** Adds a dividing line below the items. Do not use on divided lists */
    isDivided: PropTypes.bool,
    /** For internal use: adds a dividing line below all items but the last. isDivided set to true will overwrite this param */
    isDividedList: PropTypes.bool,
    /** A function to be called if the item is clicked */
    onClick: PropTypes.func
};

ListItem.defaultProps = {
    children: null,
    isSelected: false,
    isDivided: false,
    isDividedList: false,
    onClick: null
};

export default ListItem;
