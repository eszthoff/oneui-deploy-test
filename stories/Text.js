import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { boolean, text, withKnobs } from '@storybook/addon-knobs'; // eslint-disable-line import/no-extraneous-dependencies
import Text from '../src/components/Text';

storiesOf('Text', module)
    .addDecorator(withKnobs)
    .add('Text', () => (
        <Text inline={boolean('Inline text', false)} muted={boolean('Muted text', false)}>
            {text('Content', 'This is some text content')}
        </Text>
    ));
