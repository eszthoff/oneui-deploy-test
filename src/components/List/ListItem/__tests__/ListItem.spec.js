import React from 'react';
import toJson from 'enzyme-to-json';
import ListItem from '../ListItem';
import Text from '../../../Text';

describe('ListItem component', () => {
    it('should render ListItem correctly', () => {
        const wrapper = shallow(
            <ListItem>
                <Text>An item</Text>
            </ListItem>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should turn string items to Text', () => {
        const wrapper = mount(<ListItem>Simple string</ListItem>);

        expect(wrapper.find('Text')).toHaveLength(1);
    });
    it.skip('should not add clickable class when onClick is not defined', () => {
        // This test is failing due to bug in bem. See https://jira.textkernel.nl/browse/ONEUI-34
        const wrapper = mount(<ListItem>An item</ListItem>);

        expect(wrapper.find('.ListItem--clickable')).toHaveLength(0);
    });
    it('should add clickable class when onClick is defined', () => {
        const wrapper = mount(<ListItem onClick={jest.fn()}>An item</ListItem>);

        expect(wrapper.find('.ListItem--clickable')).toHaveLength(1);
    });
});
