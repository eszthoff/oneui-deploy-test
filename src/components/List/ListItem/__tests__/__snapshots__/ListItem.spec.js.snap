// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component should render ListItem correctly 1`] = `
<li
  className="ListItem ListItem--clickable"
>
  <div
    className="ListItem__container"
    onClick={null}
    role="presentation"
  >
    <Text
      inline={false}
      key=".0"
      muted={false}
      size="normal"
    >
      An item
    </Text>
  </div>
</li>
`;
